api/v1alpha1/hosted_controlplane.go

```diff
@@ -25,12 +26,12 @@ type HostedControlPlane struct {
 
 // HostedControlPlaneSpec defines the desired state of HostedControlPlane
 type HostedControlPlaneSpec struct {
-	ReleaseImage string                      `json:"releaseImage"`
-	PullSecret   corev1.LocalObjectReference `json:"pullSecret"`
-	IssuerURL    string                      `json:"issuerURL"`
-	ServiceCIDR  string                      `json:"serviceCIDR"`
-	PodCIDR      string                      `json:"podCIDR"`
-	MachineCIDR  string                      `json:"machineCIDR"`
+	ReleaseImage   string                      `json:"releaseImage"`
+	PullSecret     corev1.LocalObjectReference `json:"pullSecret"`
+	IssuerURL      string                      `json:"issuerURL"`
+	MachineNetwork MachineNetworkEntries       `json:"machineNetwork"`
+	ClusterNetwork ClusterNetworkEntries       `json:"clusterNetwork"`
+	ServiceNetwork ipnet.IPNets                `json:"serviceNetwork"`
```

Comment:
naive question: why `ServiceNetwork` and `MachineNetwork` are defined differently? why  `ServiceNetwork` does not need *Entries but `IPNets` directly? isn't `MachineNetworkEntry` effectively the same than `ipnet.IPNets`?

Reply:
Good question. I just made it match the install config, but that's not necessarily the ideal structure (since it's impermanent).

ClusterNetworkEntry definitely needs additional data. And as Clayton observed, every time we add a single-value array, we cry (paraphrasing here) https://github.com/openshift/installer/pull/2829/files#r359005851